lib/matplotlib/mpl_gui/_manage_backend.py

```diff
@@ -0,0 +1,174 @@
+from matplotlib.backends import backend_registry
+
+
+_backend_mod = None
+
+_log = logging.getLogger(__name__)
+
+
+def current_backend_module():
+    """
+    Get the currently active backend module, selecting one if needed.
+
+    Returns
+    -------
+    matplotlib.backend_bases._Backend
+    """
+    if _backend_mod is None:
+        select_gui_toolkit()
+    return _backend_mod
+
+
+def select_gui_toolkit(newbackend=None):
+    """
+    Select the GUI toolkit to use.
+
+    The argument is case-insensitive.  Switching between GUI toolkits is
+    possible only if no event loop for another interactive backend has started.
+    Switching to and from non-interactive backends is always possible.
+
+    Parameters
+    ----------
+    newbackend : Union[str, _Backend]
+        The name of the backend to use or a _Backend class to use.
+
+    Returns
+    -------
+    _Backend
+       The backend selected.
+
+    """
+    global _backend_mod
+
+    # work-around the sentinel resolution in Matplotlib 😱
+    if newbackend is None:
+        newbackend = dict.__getitem__(rcParams, "backend")
+
+    if newbackend is rcsetup._auto_backend_sentinel:
+        current_framework = cbook._get_running_interactive_framework()
+        mapping = {
+            "qt": "qtagg",
+            "gtk3": "gtk3agg",
+            "gtk4": "gtk4agg",
+            "wx": "wxagg",
+            "tk": "tkagg",
+            "macosx": "macosx",
+            "headless": "agg",
+        }
+
+        best_guess = mapping.get(current_framework, None)
+        if best_guess is not None:
+            candidates = [best_guess]
+        else:
+            candidates = []
+        candidates += ["macosx", "qt5agg", "gtk3agg", "tkagg", "wxagg"]
+
+        # Don't try to fallback on the cairo-based backends as they each have
+        # an additional dependency (pycairo) over the agg-based backend, and
+        # are of worse quality.
+        for candidate in candidates:
+            try:
+                return select_gui_toolkit(candidate)
+            except ImportError:
+                continue
+
+        else:
+            # Switching to Agg should always succeed; if it doesn't, let the
+            # exception propagate out.
+            return select_gui_toolkit("agg")
+
+    if isinstance(newbackend, str):
+        # Backends are implemented as modules, but "inherit" default method
+        # implementations from backend_bases._Backend.  This is achieved by
+        # creating a "class" that inherits from backend_bases._Backend and whose
+        # body is filled with the module's globals.
+
+        backend_name = backend_registry.resolve_gui_or_backend(newbackend)[0]
+        print(backend_name)
+        mod = importlib.import_module('matplotlib.backends.backend_' + backend_name)
+        if hasattr(mod, "Backend"):
+            orig_class = mod.Backend
+
+        else:
+            class orig_class(matplotlib.backend_bases._Backend):
+                locals().update(vars(mod))
```

Comment:
This is where I hacked, because mpl_gui was not consistent with the changes made by @ianthomas23 to the backend registry. I somewhat suspect a lot of this file could change, but am uncomfortable enough with all of this that I haven't tried yet.  I can try, but maybe it's an easy skim by folks who understand this better.